src/player.rs

```diff
@@ -48,26 +50,85 @@ pub enum PlayerEvent {
     EndOfStream,
     MetadataUpdated(Metadata),
     StateChanged(PlaybackState),
+    FrameUpdated,
     Error,
 }
 
 #[derive(Clone)]
+pub struct Frame {
+    width: i32,
+    height: i32,
+    // TODO:
+    // buffer: gst::Buffer
```

Comment:
Maybe we can store a sample? Then there wouldn't be a need to store width and height, perhaps. Anyway just a nit.

Reply:
Then we would have to keep two copies of the buffer around all the time, once in the Vec and another time in the buffer inside the sample. I was going to replace this with a sample once I implemented rendering via GL, or at least without the copy into a Vec.